static/js/server_events_dispatch.js

```diff
@@ -657,7 +657,11 @@ export function dispatch_normal_event(event) {
                     }
                     break;
                 case "read":
-                    unread_ops.process_read_messages_event(event.messages);
+                    if (event.op === "add") {
+                        unread_ops.process_read_messages_event(event.messages);
+                    } else {
+                        unread_ops.process_unread_messages_event(event.messages);
+                    }
```

Comment:
We probably want to have the code path for "receive an mark-as-unread" in a separate commit from the UI to mark messages as unread, since we'll be able to merge that as soon as we've finalized the API without finalizing UI decisions.

Reply:
Yeah, I'm not even close to splitting up commits until some of the UI decisions get settled.  My basic plan is this:

* handle (theoretical) incoming events
* add one single way to un-read messages
* add next way
* add next way
* add way to resume reading

But it's still in flux.

The one clear commit to break out is the backend changes.